types & grammar/ch3.md

```diff
@@ -388,29 +388,29 @@ Object.getOwnPropertySymbols( a );
 // [ Symbol(my own symbol) ]
 ```
 
-While symbols are not actually private (`Object.getOwnPropertySymbols(..)` reflects on the object and reveals the symbols quite publicly), using them for private or special properties is likely their primary use-case. For most developers, they may take the place of property names with `_` underscore prefixes, which are almost always by convention signals to say, "hey, this is a private/special/internal property, so leave it alone!"
+Apesar de symbols não serem realmente privados (`Object.getOwnPropertySymbols(..)` reflete no objeto e revela os publicamente quase publicamente), usar eles para criar propriedades privadas é provavelmente o seu uso primário. Para a maioria dos desenvolvedores, eles podem tomar o lugar de nomes de propriedade com `_` underscore prefixes, que é quase sempre uma convenção para dizer, "hey, esta é uma propriedade privada/especial/interna, então não mexa nela!"
 
-**Note:** `Symbol`s are *not* `object`s, they are simple scalar primitives.
+**Nota:** `Symbol`s *não* são `objects`s, eles são simples primitivos escalares.
 
-### Native Prototypes
+### Prototypes Nativos
 
-Each of the built-in native constructors has its own `.prototype` object -- `Array.prototype`, `String.prototype`, etc.
+Cada um dos construtores nativos têm seu próprio objeto `.prototype` -- `Array.prototype`, `String.prototype`, etc.
 
-These objects contain behavior unique to their particular object subtype.
+Estes objetos contém comportamento únicos para seu subtipo de objeto específico.
 
-For example, all string objects, and by extension (via boxing) `string` primitives, have access to default behavior as methods defined on the `String.prototype` object.
+Por exemplo, todos os objetos strings, e por extensão (via boxing) `string` primitivos, têm acesso ao comportamento padrão como métodos definidos no objeto `String.prototype`.
 
-**Note:** By documentation convention, `String.prototype.XYZ` is shortened to `String#XYZ`, and likewise for all the other `.prototype`s.
+**Nota:** Por convenção, `String.prototype.XYZ` é encurtado para `String#XYZ`, e da mesma todos os outros `.prototype`s.
 
-* `String#indexOf(..)`: find the position in the string of another substring
-* `String#charAt(..)`: access the character at a position in the string
-* `String#substr(..)`, `String#substring(..)`, and `String#slice(..)`: extract a portion of the string as a new string
-* `String#toUpperCase()` and `String#toLowerCase()`: create a new string that's converted to either uppercase or lowercase
-* `String#trim()`: create a new string that's stripped of any trailing or leading whitespace
+* `String#indexOf(..)`: encontra a posição na string de outra substring
+* `String#charAt(..)`: acessa o caractere em uma posição na string
+* `String#substr(..)`, `String#substring(..)`, and `String#slice(..)`: extrai uma parte da strsing como uma nova string
```

Comment:
`String#substr(..)`, `String#substring(..)`, ~and~**e** `String#slice(..)` (...) ~strsing~**string**